kratos/includes/define_python.h

```diff
@@ -103,15 +102,53 @@ PYBIND11_DECLARE_HOLDER_TYPE(T, Kratos::intrusive_ptr<T>);
 #define KRATOS_REGISTER_IN_PYTHON_FLAG(module,flag) \
     KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION(module,flag);
 
-// This function is used to print the ofstream-operator
-// i.e. printing an object will give the same result in Python as in C++
-// To be defined as the "__str__" function
-// e.g. ".def("__str__", PrintObject<ProcessInfo>)"
-// It replicates the function "self_ns::str(self))" of boost-python
+/** 
+ * @brief This function is used to print the ofstream-operator
+ * @details i.e. printing an object will give the same result in Python as in C++
+ * To be defined as the "__str__" function
+ * e.g. ".def("__str__", PrintObject<ProcessInfo>)"
+ * It replicates the function "self_ns::str(self))" of boost-python
+ */
 template< class T>
 std::string PrintObject(const T& rObject)
 {
     std::stringstream ss;
     ss << rObject;
     return ss.str();
 }
+
+/**
+ * @brief Converts a vector to a Python list using pybind11.
+ * @details This function is generic enough to be moved to a more general place.
+ * @tparam T The type of the vector.
+ * @param results The vector to convert.
+ * @return The converted Python list.
+ */
+template<typename T>
+pybind11::list VectorToPyList(const T& results) {
+    pybind11::list list_results;
+    for (auto& r_result : results) {
+        list_results.append(r_result);
+    }
+    return list_results;
+}
```

Comment:
[`std::vector` and other standard containers are implicitly converted to a python list if you `#include "pybind11/stl.h"`](https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html)

By the way, `results` is passed as a **const** reference `const T& results` but is iterated on by mutable reference `for (auto& r_results : results)`.

Reply:
also our vectors and matrices follow the buffer protocol, so it should be possible to convert them to list by simply doing 

~~~py
list(kratos_vector)
~~~

if you want that conversion to be automatic, than the way of doing it should be this one

https://github.com/KratosMultiphysics/Kratos/blob/5f204140be9345b9cb1c5420539415ca6f1556c7/kratos/python/add_vector_to_python.cpp#L248

Reply:
I will give a try